src/Raven.Client/Util/AtomicDictionary.cs

```diff
@@ -29,6 +29,9 @@ public AtomicDictionary(IEqualityComparer<string> comparer)
             _locks = new ConcurrentDictionary<string, object>(comparer);
         }
 
+        /// <summary>
+        /// This locks the entire dictionary. Use carefully.
+        /// </summary>
         public IEnumerable<TVal> Values => _items.Values;
```

Comment:
Are there any usages of this, if not, just kill the prop.
Also, I think we can just use `ConcurrentDictionary<string,Lazy<T>>` here instead, no?